aten/src/ATen/native/TensorShape.cpp

```diff
@@ -458,7 +470,13 @@ Tensor select(const Tensor& self, int64_t dim, int64_t index) {
   auto storage_offset = self.storage_offset() + index * strides[dim];
   sizes.erase(sizes.begin() + dim);
   strides.erase(strides.begin() + dim);
-  return self.as_strided(sizes, strides, storage_offset);
+  auto result = self.as_strided(sizes, strides, storage_offset);
```

Comment:
It seems like as_strided should accept names, and then the rest of the code would just look like the existing code.
Will the reduction_op function be used for something else that it makes sense to break it out?

Reply:
> It seems like as_strided should accept names, and then the rest of the code would just look like the existing code.

That's an interesting proposal. `as_strided` can do pretty crazy things to tensors (change their sizes and strides). In a named-tensor-world, the user shouldn't use `as_strided` directly because it shouldn't be necessary. Because of that, I don't want to add a `names` argument to `as_strided`.

> Will the reduction_op function be used for something else that it makes sense to break it out?

It'll be useful for other functions that take an `int dim`.